python/pyspark/pandas/indexes/base.py

```diff
@@ -1907,6 +1908,9 @@ def append(self, other: "Index") -> "Index":
             )
 
         index_fields = self._index_fields_for_union_like(other, func_name="append")
+        # Since pandas 1.5.0, the order of category matters.
+        if isinstance(other, CategoricalIndex):
+            other = other.reorder_categories(self.categories.to_list())
```

Comment:
shall we check the version, and only call `reorder_categories` when version >= 1.5.0?

or this `reorder_categories` also work with 1.4.x?

Reply:
Oh, yeah we should separate the test. Let me address it

Reply:
Since we should follow behavior of latest pandas, let me leave the code as it is.

Let me separate the test.

Reply:
We may also have a separate `append` for CategoricalIndex to maintain clearer abstraction since we only make `MultiIndex` an exception (allow different behaviors for MultiIndex) in this file.

Reply:
Considering the code change is minimum, feel free to keep the current implementation if you prefer :)

Reply:
Sounds good, @xinrong-meng . Thanks for the suggestion.

Let me keep it as it is for now, and will do the refine the structure after fixing all other unmatched behavior!